### PR TITLE
[SPARK-49999][PYTHON][CONNECT] Support optional "column" parameter in box, kde and hist plots

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Support for the optional “column” parameter has been added in box, kde, and hist plots. Now, when the column is not provided, all columns of valid types (NumericType, DateType, TimestampType) will be used to build the plots.

### Why are the changes needed?
- Reach parity with Pandas (on Spark) default behavior.
- Simplify usage by reducing the need for explicitly specifying columns.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Unit tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by Pshak-20000** [comment]
Hi ,

I want to contribute to the project and can help out. Please let me know what to do!

Thanks!

File: python/pyspark/sql/plot/plotly.py [hunk 1]
```diff
@@ -214,3 +209,35 @@ def plot_histogram(data: "DataFrame", **kwargs: Any) -> "Figure":
...
+    """
+    Processes the provided column parameter for a DataFrame.
+    - If `column` is None, returns a list of numeric or datetime columns from the DataFrame.
+    - If `column` is a string, converts it to a list first.
+    - If `column` is a list, it checks if all specified columns exist in the DataFrame
+      and are of valid types (NumericType, DateType, or TimestampType).
+    - Raises a PySparkTypeError if any column in the list is not present in the DataFrame
+      or has an invalid type.
+    """
+    valid_types = (NumericType, DateType, TimestampType)
```
Comment (zhengruifeng) [inline python/pyspark/sql/plot/plotly.py:224]: Does these plot actually accept `DateType` and `TimestampType`?

it seems we only support `NumericType` in the PS side?
https://github.com/apache/spark/blob/54edfd3cdb1b2325b04734c3b9bdc289598de558/python/pyspark/pandas/plot/plotly.py#L156-L160

**Comment by xinrong-meng** [comment]
Hi @Pshak-20000 thank you for reaching out!

A good starting point would be to look at the open issues on the [Apache Spark JIRA](https://issues.apache.org/jira/projects/SPARK/issues).That will give you a good idea of the areas you can contribute to.

Also, participating in code reviews is a great way to get familiar with the codebase.
You can find more detailed guidance on https://spark.apache.org/contributing.html good luck with your journey into Spark contributions!

File: python/pyspark/sql/plot/plotly.py (hunk 1 above)
Comment (xinrong-meng) [reply python/pyspark/sql/plot/plotly.py:224]: Good catch!
Box plots seem to support NumericType only.
Density and histogram plots will accept `np.datetime64` in PS, as https://github.com/apache/spark/blob/master/python/pyspark/pandas/plot/core.py#L142.

Comment (xinrong-meng) [reply python/pyspark/sql/plot/plotly.py]: Modified to accept NumericType data only, thank you!

**Comment by Pshak-20000** [comment]
Hi @xinrong-meng ,

Thanks for your reply! I'm new here, and I’m eager to contribute. I’ll check out the open issues on the Apache Spark JIRA and the contributing guide.

If there’s anything specific I can help with, please let me know. I’d really appreciate any guidance!

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/plot/plotly.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/plot/plotly.py]: let's also adjust the PR description

**Comment by xinrong-meng** [comment]
Merged to master, thank you!

**Pull request closed without merging**